### ISSUE TITLE
building with Ninja

### ISSUE THREAD

**Issue by Trass3r** [rose-compiler/rose#52]
```sh
git clone --depth=100 -b develop https://github.com/rose-compiler/rose rose
cd rose
git checkout fc6bb3504961cd8cbe7d758eaa4a86069ad81589
mkdir release && cd release
cmake -GNinja -Denable-binary-analysis=yes -Denable-c=no -Denable-java=no \
        -Denable-fortran=no -Ddisable-tutorial-directory=yes \
        -Ddisable-test-directory=yes ..

$ ninja -k4 -j4
[1/854] FAILED: src/util/commandlineProcessing/CMakeFiles/util_commandlineProcessing.dir/commandline_processing.C.o
../src/util/commandlineProcessing/commandline_processing.C:2:10: fatal error: rosePublicConfig.h: No such file or directory
[126/854] Linking CXX static library lib/libRoseSQLite3xDatabase.a
ninja: build stopped: cannot make progress due to previous errors.
$ ninja -j1
[3/729] Generating ../../frontend/SageIII/AST_FILE_IO.h, [...]
cd ~/build/src/ROSETTA/src && /usr/bin/cmake -E copy_directory ~/rose/src/ROSETTA/Grammar ~/build/src/ROSETTA/Grammar && /usr/bin/cmake -E make_directory ~/build/src/frontend/SageIII/astFileIO && ~/build/bin/CxxGrammarMetaProgram ~/build/src/frontend/SageIII/
terminate called after throwing an instance of 'std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >'
```
Building with make works (also in parallel).

**Comment by sei-eschwartz** [comment]
The actual problem is that the invocation of the CxxGrammarMetaProgram fails with:

```
root@c95617ca8a3b:~/pharos/scripts/rose/release# (cd /root/pharos/scripts/rose/release/src/ROSETTA/src && /usr/bin/cmake -E copy_directory /root/pharos/scripts/rose/src/ROSETTA/Grammar /root/pharos/scripts/rose/release/src/ROSETTA/Grammar && /usr/bin/cmake -E make_directory /root/pharos/scripts/rose/release/src/frontend/SageIII/astFileIO && /root/pharos/scripts/rose/release/bin/CxxGrammarMetaProgram /root/pharos/scripts/rose/release/src/frontend/SageIII/
> )
ERROR: File not found -- ../src/ROSETTA/src/binaryInstruction.C 
```

It took me a while to figure out why this was happening.  The path to binaryInstruction.C is apparently hard-coded into the CxxGrammarMetaProgram:

```
src/ROSETTA/src/binaryInstruction.C:    CLASS_WITHOUT_Sg.setPredeclarationString("Sg" #CLASS_WITHOUT_Sg "_HEADERS", __FILE__)
src/ROSETTA/src/binaryInstruction.C:    CLASS_WITHOUT_Sg.setFunctionPrototype("Sg" #CLASS_WITHOUT_Sg "_OTHERS", __FILE__)
```

When using the normal Make generator, the compiler is invoked with absolute paths.  Using the ninja generator and in in-tree build, the compiler is invoked with relative paths, and this is enough to mess up the command invocation.

**Comment by matzke1** [comment]
Fixed by commit 55dd85e7, JIRA issue ROSE-2173. Should be released later today if all goes well.

**Comment by sei-eschwartz** [comment]
Thanks @matzke1!
